guillotina/catalog/parser.py

```diff
@@ -86,6 +86,8 @@ def __call__(self, params: typing.Dict) -> BasicParsedQueryInfo:
             path = "/" + path.strip("/")
         else:
             path = get_content_path(self.context)
+        if path.endswith("/") is False and path != "/":
+            path = path + "/"
```

Comment:
I'm not 100% sure if we want to add this trailing slash in all cases or only when the path is defined implicitly based on context.

I could see someone specifying `path__starts` explicitly without the trailing slash and expect the current behavior (items from different folders that start with same prefix)

Reply:
yes! I agree @masipcat this could be a good functionality to have, my thought though when I did this, was to preserve the same behavior when not passing `paths__starts`, which in some cases can lead to confusions on how it works. What is clear is that by calling `/db/container/foo_folder/@search` I am not expecting to get any results from another folder that is not `foo_folder`. 
Do you think that is worthy to have this dual behavior when passing `path__starts`? If we all agree, we just have to specify it somewhere, or it can lead to miss behaviors.

Reply:
I wouldn't say it's a dual behavior. We are not changing how the filter `path__starts` behaves. We are changing how the implicit search in a folder works. And as you said, it's clear that the current behavior of `@search` in a folder doesn't work as expected.

But I'd like to know what @bloodbare and @vangheem think about this as well

Reply:
Also it's funny the code in L86 because right now the trailing slash is always stripped. I'm not sure if this behavior is intended or should've been `path = "/" + path.lstrip("/")`

Reply:
You are right @masipcat . If you all agree and want to have the same behavior as now, but just adding the slash when using the context, it is fine for me, however, I think that the trailing slash being removed when injecting path__starts on the query for me is not the behavior the user expects. I think that what you propose with the line `path = "/" + path.lstrip("/")` is the way to go. We let the user choose how he wants to search.

Reply:
@masipcat the last commit does what we have talked X)